### PR TITLE
Document & Test (partial).

### PR DESCRIPTION
**Pull request by kevindecapite**
Refs:

- https://github.com/cyjoelchen/php-sweph/issues/9
- https://github.com/cyjoelchen/php-sweph/issues/22

I am adding documentation one function at a time, using the POD method prescribed by @aloistr and simultaneously adding a test for each function as well.

This PR is a WIP (work-in-progress) and should not be merged until all current methods are documented and tested.

Collaboration is welcome.

### CODE REVIEW

File: build-se [hunk 1]
```diff
@@ -5,8 +5,8 @@ phpize && \
 make && \
 make install && \
 echo "extension=sweph.so" > /usr/local/etc/php/conf.d/sweph.ini &&
+yes n | make test && \
```
Comment (kevindecapite) [inline build-se:8]: Automatically run tests on each build and do not ask to report results to the PHP community.

File: tests/swe_calc_pctr.phpt [hunk 2]
```diff
@@ -0,0 +1,31 @@
+--TEST--
+Basic test
+--SKIPIF--
+<?php
+# @todo Determine why results are different in PHP CLI vs. tests.
```
Comment (kevindecapite) [inline tests/swe_calc_pctr.phpt:5]: Test output is:

```
array(8) {
  [0]=>
  float(1.2862540711761E+248)
  [1]=>
  float(1.9687986489104E-28)
  [2]=>
  float(2.5489505080981E+179)
  [3]=>
  float(3.9286866178939E-315)
  [4]=>
  float(0)
  [5]=>
  float(0)
  ["serr"]=>
  string(48) "barycentric Moshier positions are not supported."
  ["rc"]=>
  int(-1)
}
```

Output from PHP CLI is the expectation in the test. Check locally with:

```
> php -a
php > var_dump(swe_calc_pctr(2452275.5, 1, 2, SEFLG_MOSEPH));
```

File: tests/swe_fixstar.phpt [hunk 3]
```diff
@@ -12,24 +12,24 @@ swe_set_ephe_path(null);
...
+  float(-0.0077695918085574)
   ["star"]=>
-  string(7) "Polaris"
+  string(13) "Polaris,alUMi"
   ["serr"]=>
-  string(77) "SwissEph file 'sefstars.txt' not found in PATH '.:/users/ephe2/:/users/ephe/'"
+  string(0) ""
   ["rc"]=>
-  int(-1)
+  int(4)
```
Comment (kevindecapite) [inline tests/swe_fixstar.phpt:34]: How were we getting different results running under the same conditions?

**Comment by aloistr** [comment]
The test does not set ephepath, so it defaults to the old internal default path
".:/users/ephe:/users/ephe2"

I have the files in this path, you probably don't

That exposes a general problem for tests.
The internally referenced ephemeris files must exist and be found in the test environment.

**Comment by aloistr** [comment]
Having a copy of file sefstars.txt in directory php-sweph solves the issue, on Linux at least.
The default path contains the dot directory .

We need files seas_18.se1  sefstars.txt  semo_18.se1  sepl_18.se1 to make swe_calc test use Swiss Ephemeris mode and and Moshier mode.

**Comment by kevindecapite** [comment]
> The internally referenced ephemeris files must exist and be found in the test environment.

I thought about including them in this repo but considered it might be acceptable to allow the function to fail as I originally had it since these tests only need to verify that the SE methods are implemented correctly.

I am open to the idea of including a minimal set of the data files in this repo, however.

**Comment by aloistr** [comment]
I have added the necessary 4 files 
seas_18.se1  sefstars.txt  semo_18.se1  sepl_18.se1

Test cases should have dates in the year range 1800 - 2399.
Moshier mode is deprecated, it will go in future releases.
Built in binary ephemeris files will replace it, to support single file installations.

**Comment by aloistr** [comment]
float or double
Kevin, you converted my 'double' in pods to 'float'

I suggest to stick to 'double'.
I know that the formal name of floating point number in PHP is 'float', but on many places
it is said  "Floating point numbers (also known as "floats", "doubles", or "real numbers")"

We are dealing with an API to a C library here, and in C "float" means reduced precision 4-byte real number, "double" means ful precision 8-byte real number.
As a C programmer it confuses me to see 'float'.

As double is widely accepted as equivalent to float, in PHP, I suggest we use 'double' in the docu.

**Comment by kevindecapite** [comment]
> I suggest to stick to 'double'.

I did expect that change to not go unnoticed. I was keeping with common PHP syntax as rarely, if ever, will you see type definitions as `double`. I am fine to use that term, however, as it's more correct from a broader technological standpoint (i.e. outside the PHP community).

I only meant to not confuse the average PHP developer with it.

**Review by aloistr — APPROVED** [review]
OK, merge all docu_test to master

**Comment by aloistr** [comment]
Test tests/swe_get_library_path.phpt wil always fail for me, and probably for others too.

my diff is
001+ string(38) "/home/alois/php-sweph/modules/sweph.so"
001- string(32) "/root/php-sweph/modules/sweph.so"
and it will never succeed, because not all users will install php-sweph in /root

I suggest the test should only check that "/php-sweph/modules/sweph.so" is contained in the reply.

**Comment by kevindecapite** [comment]
> Test tests/swe_get_library_path.phpt wil always fail for me, and probably for others too.

Thank you for pointing that out. I changed the test to use a substring of the entire path (`/modules/sweph.so`) so this should pass in all environments now.

**Comment by kevindecapite** [comment]
@aloistr 

I deleted this branch now that it's merged. This is a typical part of the workflow. I am leaving the issue open, however, as there is more to do still.

To update and clean your local environment:

```
> git fetch
> git check master
> git pull origin master
> git remote prune origin
```

I like to remove my local copies of deleted remote branches with:

```
> git branch -d name_of_branch
```

**Pull request merged**